### DISCUSSION TITLE
Cutting Release Candidate

### DISCUSSION THREAD

**Discussion by matmair** [inventree/InvenTree#7810] [Ideas]
@SchrodingersGat there are a lot of changes in 0.16.x already - can we cut a release candidate `0.16.0rc0` to run CI and manual tests on? I would like to manually test SSO changes for their robustness in an OSWAP test suite. It would be easier for my CI if the release were on the main org as that ensures it is the same regiment as with the official images

**Comment by SchrodingersGat** [comment]
Absolutely. I'll start a `0.16.x` branch and start migrating issues which won't make it to `0.17.x`. Then we should make a push to close out any urgent ones (and we can backport to `0.16.x`)

**Reply by SchrodingersGat** [reply]
0.16.x branch is out now: https://github.com/inventree/InvenTree/tree/0.16.x

**Reply by matmair** [reply]
Can I cut the first release candidate or do you want to do that?

**Reply by SchrodingersGat** [reply]
Happy for you to do that

**Comment by wolflu05** [comment]
@matmair I think it would make sense to manually adjust the auto generated changelog a bit. E.g.
- the alpine version bump was reverted as far as I remember, but still shows up in the breaking changes
- in #7684 I wrote a small text how the icon migration works. As this is a mandetory step if icons were used previously, I think this should definitely be added to the top of the changelog.

**Reply by matmair** [reply]
I would ignore the changelog of the rc, it has to be rewritten for the full release anyhow with the current processes and as written in the very top this should not be used for production

**Reply by wolflu05** [reply]
Ok. Then this is just a proposal for the 0.16 changelog.

**Comment by wolflu05** [comment]
@matmair BTW I just found out that the docker stable tag got overwritten by 0.16 rc 0, was this intentional?

**Reply by SchrodingersGat** [reply]
It should not have been. We will have to adjust our scripts to not push a stable release for "rc" versions.

**Comment by SchrodingersGat** [comment]
@matmair this has overwritten our "stable" tags, which is problematic for both the installer and the docker image.

To fix this I think we have to:

1. Remove this release and tag entirely
2. Make a new release on the `0.15.x` branch (which will push a new stable tag)
3. Update our release script to *not* push to stable on an `rc` tag
4. Make a new release candidate

Are you happy with this?

**Reply by matmair** [reply]
I think it should work without publishing a new 0.15.x release, I will try this now

**Reply by matmair** [reply]
It seems to work, 0.15.7 is back in stable as the newest build for packages. Docker will take 3~ hours to build

**Reply by SchrodingersGat** [reply]
Ok, well that's something. Thanks!

**Reply by wolflu05** [reply]
But can we please release 0.15.8 before we release 0.16?

**Reply by SchrodingersGat** [reply]
Yes, I can do that. Is there anything else critical we need to back-port?

**Reply by wolflu05** [reply]
For me nothing really critical. It's just the backported plugin_static template tag so that my plugins will work with 0.15.x and also with 0.16.x without any version conditions doing things differently for 0.15.
